### ISSUE TITLE
kube-state-metrics process generates uninteresting metrics about itself with go_* prefix

### ISSUE THREAD

**Issue by ahakanbaba** [kubernetes/kube-state-metrics#20]
If you look at the output of the curl (mentioned in the README.md ) the start of the output has these: 

go_gc_duration_seconds{quantile="0"} 6.1791e-05
go_gc_duration_seconds{quantile="0.25"} 6.3312e-05
go_gc_duration_seconds{quantile="0.5"} 0.000103029
go_gc_duration_seconds{quantile="0.75"} 0.000126511
go_gc_duration_seconds{quantile="1"} 0.000670194
go_gc_duration_seconds_sum 0.001212694
go_gc_duration_seconds_count 7
//  HELP go_goroutines Number of goroutines that currently exist.
// TYPE go_goroutines gauge
go_goroutines 23

As far as I understand these are go framework metrics about the kube-state-metrics process, which are quite uninteresting. 

The reason of that should be the prometheus.UninstrumentedHandler() function call. 
If I am not mistaken that handler by default adds metrics about the process. 

The definition says this: 
// UninstrumentedHandler returns an HTTP handler for the DefaultGatherer.
//
// Deprecated: Use promhttp.Handler instead. See there for further documentation.
func UninstrumentedHandler() http.Handler {

This issue could be resolved by using promHttp handler and separating the metrics that start with go_ from the rest.  (I know this is a vague explanation. After some more investigation,  I can add more details about the proposed impl changes to this issue. ) 

We would like to improve the kube-state-metrics in that regard. 
The primary purpose of this issue is to communicate this to other developers and avoid replicated work.

**Comment by grobie** [comment]
These are only very few timeseries. What are your issues with exporting these?

**Comment by ahakanbaba** [comment]
These are metrics

**Comment by ahakanbaba** [comment]
Hi all for the comments, suggestions and insights. 
We have implemented a filter agent in front of the kube-state-metrics that purges the go_* metrics. 
So at this point, for us this pull request it not crucial.

We do not want these go_* metrics because

We do not need them.
Once they are in the metrics store database, they require some cognitive activity to discard and remember what they are. We have seen that they make browsing the saved metrics more complex.
They also consume small but non-zero space.
Since we actively remove the go_* metrics ourselves, we are abandoning this pull request. 
We still believe that retaining the go_* metrics in the kube-state-metrics defeats the Keep It Simple and You Aren't Gonna Need It principles. Preferably in the future the prometheus golang client would not add these metrics by default.

Thanks everyone for their time.

SImilarly the relevant pull request  #21  has also been closed.
